src/main/java/uk/gov/cshr/controller/IdentityController.java

```diff
@@ -1,18 +1,121 @@
+import uk.gov.cshr.domain.Identity;
+import uk.gov.cshr.domain.Role;
 import uk.gov.cshr.dto.IdentityDTO;
+import uk.gov.cshr.repository.IdentityRepository;
+import uk.gov.cshr.service.AuthenticationDetails;
+import uk.gov.cshr.service.IdentityService;
+import uk.gov.cshr.service.RoleService;
 import uk.gov.cshr.service.security.IdentityDetails;
 
-@RestController
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Controller
+@RequestMapping("/management/")
 public class IdentityController {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(RoleController.class);
+
+    @Autowired
+    private IdentityService identityService;
+
+    @Autowired
+    private RoleService roleService;
+
+    @Autowired
+    private IdentityRepository identityRepository;
+
+    @Autowired
+    private AuthenticationDetails authenticationDetails;
+
     @RequestMapping(value = "/identity", method = RequestMethod.GET)
     public IdentityDTO getIdentityDetailsfromAccessToken(Authentication authentication) {
         IdentityDetails identityDetails = (IdentityDetails) authentication.getPrincipal();
         return new IdentityDTO(identityDetails.getIdentity());
     }
+
+    @GetMapping("/identities")
+    public String identities(Model model) {
+        LOGGER.info("Listing all identities");
+
+        Iterable<Identity> identities = identityService.findAll();
+        model.addAttribute("identities", identities);
+        return "identityList";
+    }
+
+    @GetMapping("/identities/update/{uid}")
+    public String identityUpdate(Model model,
+                                 @PathVariable("uid") String uid) {
+
+        LOGGER.info("{} editing identity for uid {}", authenticationDetails.getCurrentUsername(), uid);
+
+        Optional<Identity> optionalIdentity = identityService.getIdentity(uid);
+        Iterable<Role> roles = roleService.findAll();
+
+        if (optionalIdentity.isPresent()) {
+            Identity identity = optionalIdentity.get();
+            model.addAttribute("identity", identity);
+            model.addAttribute("roles", roles);
+            return "updateIdentity";
+        }
+
+        LOGGER.info("No identity found for uid {}", authenticationDetails.getCurrentUsername(), uid);
+        return "redirect:/management/identities";
+    }
+
+    @PostMapping("/identities/update")
+    public String identityUpdate(@RequestParam(value = "active", required = false) Boolean active, @RequestParam(value = "roleId", required = false) ArrayList<String> roleId, @RequestParam("uid") String uid) {
+
+        // get identity to edit
+        Optional<Identity> optionalIdentity = identityService.getIdentity(uid);
+
+        if (optionalIdentity.isPresent()) {
+            Identity identity = optionalIdentity.get();
+
+            Set<Role> roleSet = new HashSet();
+            // create roles from id
+            if (roleId != null) {
```

Comment:
OPTIMIZE - Can we not use `roleId.forEach`?

Reply:
lambda expression is not superior than using enhanced for loop except with a parallel stream.

Reply:
Nothing to do with speed, readability and should prevent null checks too?

Reply:
Does not prevent null checks, exception thrown on nulls. As to readability, it's a pretty standard way of iteration?